sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs

```diff
@@ -96,19 +98,8 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
                     bufferedStream.Position = 0;
                     message.Response.ContentStream = bufferedStream;
                 }
-                // We dispose stream on timeout so catch and check if cancellation token was cancelled
-                catch (ObjectDisposedException ex)
-                {
-                    ThrowIfCancellationRequestedOrTimeout(oldToken, cts.Token, ex, networkTimeout);
-                    throw;
-                }
-                // We dispose stream on timeout so catch and check if cancellation token was cancelled
-                catch (IOException ex)
-                {
-                    ThrowIfCancellationRequestedOrTimeout(oldToken, cts.Token, ex, networkTimeout);
-                    throw;
-                }
-                catch (OperationCanceledException ex)
+                // We dispose stream on timeout or user cancellation so catch and check if cancellation token was cancelled
+                catch (Exception ex) when (ex is ObjectDisposedException or IOException or OperationCanceledException or NotSupportedException)
```

Comment:
Super nit: any option to break this into multiple lines for easier reading/maintenance?